app/models/owner.js

```diff
@@ -63,25 +64,23 @@ export default Model.extend({
   },
 
   fetchBetaMigrationRequests() {
-    return this.ajax.getV3(`/user/${this.accounts.user.id}/beta_migration_requests`).then(data => {
-      this.store.pushPayload('beta-migration-request', data);
-      return this.store.peekAll('beta-migration-request');
-    });
+    return this.fetchBetaMigrationRequestsTask.perform();
   },
 
-  migrationBetaRequests: computed(function () {
-    return this.store.peekAll('beta-migration-request').filter(request =>
-      request.owner === this || request.organizations.includes(this)
-    );
-  }).volatile(),
+  fetchBetaMigrationRequestsTask: task(function* () {
+    const data = yield this.ajax.getV3(`/user/${this.accounts.user.id}/beta_migration_requests`);
+    this.store.pushPayload('beta-migration-request', data);
+    return this.store.peekAll('beta-migration-request');
+  }),
```

Comment:
Nice, I think switching this to an ember-concurrency task makes a lot of sense :+1:

Reply:
Yeah, `volatile` properties don't recalculate when `store` changes, so it caused issues. It was a weird decision from the beginning 😆

Reply:
Haha yeah I've been there for sure. You write something in a way that makes sense in the moment, and then later look back and go "Why on Earth did I do it this way?" :laughing: